src/lua/LuaGame.cpp

```diff
@@ -738,6 +739,92 @@ static int l_game_get_parts_from_date_time(lua_State *l)
 	return 6;
 }
 
+// Helper functions for the generic l_game_get_config and l_game_set_config
+// template functions since C++ does not have a "static_if" yet.
```

Comment:
That's called `if constexpr`. `std::decay` is helpful here as it strips reference and cv-qualification from the variable type.

```c++
if constexpr (std::is_same_v<std::decay_t<T>, bool>) {
/// ...
} else if constexpr (...) {
// ...
}
```

Note that you cannot `static_assert(false)` in the `else` branch [until C++23](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2593r0.html) (because Reasons).

Reply:
Fixed in 9be214d9202f6d4af143e477ace2132461f265ca

Thank you for sharing the "if constexpr" approach; learned something new :)

I did quite a bit of research but could not find that approach mentioned anywhere. The closest was a "static_if" which somebody made but required a LOT of supporting magic foo to make happen.

TBH, in retrospect I actually prefer the overloaded method approach - it's cleaner and more readable. But yes, without other code making use of them it's a lot of unnecessary code.